### PR TITLE
Add general_preprocess.py 

### PR DESCRIPTION
**Pull request by zaher88abd**
Hi, 
I relocate the file and rename it. 
Again Thank you for approving my small contributions to you greate project.

### CODE REVIEW

**Comment by wzhouuu** [comment]
I got a question here. In `create_bonnet_dataset_script.py`, seems you only delete the test set from `img_list` but not valid set. This leads to the validation set is actually a subset of training set, which based on my understanding is not correct. Validation set shouldn't include the data your model has learnt?

Please correct me if I misunderstood the code. Thanks!

**Comment by zaher88abd** [comment]
you are right @wzhouuu the validation set is a subset of training. As I know the validation set should be part of training data set, however, that depends on how the framework split the data for each iteration and this point @tano297 could explain it better than me.
